src/process_binding.cpp

```diff
@@ -0,0 +1,923 @@
+    if (decl->hasAttrs()) {
+        clang::ASTContext& ctx = decl->getASTContext();
+        for (const auto& attr : decl->attrs()) {
+            const clang::AnnotateAttr* ann =
+                clang::cast<const clang::AnnotateAttr>(attr);
+            if (ann) {
+                attrs.push_back(ann->getAnnotation().str());
+            }
+        }
+    }
+    return attrs;
+}
+
+struct indent {
+    indent(int level) : level(level) {}
+
+private:
+    friend std::ostream& operator<<(std::ostream& stream, const indent& val);
+
+    int level;
+};
+
+std::ostream& operator<<(std::ostream& stream, const indent& val) {
+    for (int i = 0; i < val.level * 4; i++) {
+        stream << " ";
+    }
+    return stream;
+}
+
+std::string strip_name_kinds(std::string s) {
+    s = pystring::replace(s, "class ", "");
+    s = pystring::replace(s, "struct ", "");
+    s = pystring::replace(s, "enum ", "");
+    s = pystring::replace(s, "union ", "");
+    return s;
+}
+
+} // namespace
+
+enum class NodeKind : uint32_t {
+    Node = 0,
+    TranslationUnit,
+    Namespace,
+    BuiltinType,
+    PointerType,
+    RecordType,
+    Parm,
+    Function,
+    Method,
+    Record,
+};
+
+std::ostream& operator<<(std::ostream& os, NodeKind k) {
+    switch (k) {
+    case NodeKind::Node:
+        os << "Node";
+        break;
+    case NodeKind::TranslationUnit:
+        os << "TranslationUnit";
+        break;
+    case NodeKind::Namespace:
+        os << "Namespace";
+        break;
+    case NodeKind::BuiltinType:
+        os << "BuiltinType";
+        break;
+    case NodeKind::PointerType:
+        os << "PointerType";
+        break;
+    case NodeKind::RecordType:
+        os << "RecordType";
+        break;
+    case NodeKind::Parm:
+        os << "Parm";
+        break;
+    case NodeKind::Function:
+        os << "Function";
+        break;
+    case NodeKind::Method:
+        os << "Method";
+        break;
+    case NodeKind::Record:
+        os << "Record";
+        break;
+    }
+    return os;
+}
+
+enum class PointerKind : uint32_t {
+    Pointer,
+    Reference,
+};
+
+enum class RecordKind : uint32_t { OpaquePtr = 0, OpaqueBytes, ValueType };
+
+using NodeId = uint64_t;
+
+struct sanitize {
+    const std::string& s;
+
+    sanitize(const std::string& s) : s(s) {}
+};
+
+std::ostream& operator<<(std::ostream& os, const sanitize& s) {
+    std::string r = s.s;
+    r = pystring::replace(r, "&", "&amp;");
+    r = pystring::replace(r, "<", "&lt;");
+    r = pystring::replace(r, ">", "&gt;");
+    r = pystring::replace(r, "\"", "&quot;");
+    return os << r;
+}
+
+struct Node {
+    std::string qualified_name;
+    NodeId id;
+    NodeId context; //< parent context (e.g. record, namespce, TU)
+    NodeKind node_kind;
+
+    Node(std::string qualified_name, NodeId id, NodeId context,
+         NodeKind node_kind)
+        : qualified_name(qualified_name), id(id), context(context),
+          node_kind(node_kind) {}
+
```

Comment:
```suggestion
virtual ~Node(){}
```
I had a warning about this at compile time.